### PR TITLE
chore: Switch to Java Text Blocks

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

Continuation of #220, making good use of:

```bash
mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-migrate-java:RELEASE -Drewrite.activeRecipes=org.openrewrite.java.migrate.lang.UseTextBlocks   
```

### CODE REVIEW

File: cloudplatform/cloudplatform-core/src/main/java/com/sap/cloud/sdk/cloudplatform/requestheader/RequestHeaderAccessor.java [hunk 1]
```diff
@@ -179,15 +179,13 @@ public static <T> T executeWithHeaderContainer(
...
-                            + " ('cds-integration-cloud-sdk'). To workaround this shortcoming, please refer to the CAP"
-                            + " documentation about how to manipulate the request context: "
-                            + "https://cap.cloud.sap/docs/java/request-contexts#defining-requestcontext.",
-                        DefaultRequestHeaderFacade.class.getName()));
+            throw new ThreadContextExecutionException(String.format("""
+                The 'executeWith...' API is currently supported only when using the Cloud SDK's 'DefaultFacade'\
+                 implementations (e.g. '%s'). This is an issue especially when using the CAP integration\
+                 ('cds-integration-cloud-sdk'). To workaround this shortcoming, please refer to the CAP\
+                 documentation about how to manipulate the request context: \
+                https://cap.cloud.sap/docs/java/request-contexts#defining-requestcontext.\
```
Comment (Johannes-Schneider) [inline cloudplatform/cloudplatform-core/src/main/java/com/sap/cloud/sdk/cloudplatform/requestheader/RequestHeaderAccessor.java:187]: These escaped line breaks are a little odd, aren't they?

Especially the ones at the last lines of text.

Comment (newtork) [reply cloudplatform/cloudplatform-core/src/main/java/com/sap/cloud/sdk/cloudplatform/requestheader/RequestHeaderAccessor.java:187]: @Johannes, what's your suggestion?

Personally I would've preferred one uninterrupted String in the first place. It makes full-text search for exception messages easier. And splitting it doesn't bring any benefit beyond readability - which could be established by editor word-wrap. (Don't know how)

File: cloudplatform/cloudplatform-connectivity/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/EnvVarDestinationLoaderTest.java [hunk 2]
```diff
@@ -83,31 +90,17 @@ void testCompleteDestinationAttributes()
...
-        final Option<String> name = httpDestination.get(DestinationProperty.NAME);
-
-        assertThat(name.get()).isEqualTo(destinationName);
-
-        final Option<ProxyType> proxyType = httpDestination.getProxyType();
-
-        assertThat(proxyType.get()).isEqualTo(ProxyType.INTERNET);
-
+        assertThat(httpDestination.get(DestinationProperty.NAME)).contains(destinationName);
+        assertThat(httpDestination.getProxyType()).contains(ProxyType.INTERNET);
```
Comment (newtork) [inline cloudplatform/cloudplatform-connectivity/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/EnvVarDestinationLoaderTest.java:99]: (Major)

What's the reason to employ manual drive-by code changes? This will increase review effort due to context switching.

File: cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java [hunk 3]
```diff
@@ -39,23 +39,18 @@
...
     // https://help.sap.com/viewer/cca91383641e40ffbe03bdc78f00f681/Cloud/en-US/df1bb55a526942b9bee78fea2ebb3162.html
-    //Mapping file extension to key store types
-    private static final Map<String, String> SUPPORTED_KEY_STORE_TYPES_AS_KEY_STORE = ImmutableMap.of(
-            "pfx", "PKCS12",
-            "p12", "PKCS12",
-            "jks", "JKS"
-    );
+    // Mapping file extension to key store types
+    private static final Map<String, String> SUPPORTED_KEY_STORE_TYPES_AS_KEY_STORE =
+        ImmutableMap.of("pfx", "PKCS12", "p12", "PKCS12", "jks", "JKS");
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java:46]: (Minor)

Not a fan. Context: I'm about to add another pair here.

File: datamodel/odata-client/src/test/java/com/sap/cloud/sdk/datamodel/odata/client/exception/ODataResponseErrorParsingTest.java [hunk 4]
```diff
@@ -173,25 +173,26 @@ void testParsingODataV2Error()
...
-                + "     }"
-                + "   ],"
-                + "  \"innererror\": {"
-                + "     \"foo\": 123,"
-                + "     \"bar\": \"ok\""
-                + "  }"
-                + "}"
-                + "}";
+        final String json = """
+            {\
```
Comment (newtork) [inline datamodel/odata-client/src/test/java/com/sap/cloud/sdk/datamodel/odata/client/exception/ODataResponseErrorParsingTest.java:177]: (Minor)

Unnecessary new-line escaping.

File: datamodel/odata-client/src/test/java/com/sap/cloud/sdk/datamodel/odata/client/request/ODataResponseComplexDataParsingTest.java [hunk 5]
```diff
@@ -85,154 +85,160 @@ public static class Person
...
+                      "Address": "187 Suffolk Ln.",
+                      "City": {
+                          "Name": "Boise",
+                          "CountryRegion": "United States",
+                          "Region": "ID"
+                      }
+                  }
+              ]
+            }
+            ]}\
```
Comment (newtork) [inline datamodel/odata-client/src/test/java/com/sap/cloud/sdk/datamodel/odata/client/request/ODataResponseComplexDataParsingTest.java:129]: (Comment/Minor)

**AFAIK** escape last new-line is not necesary - this happens implicitly, doesn't it?

File: datamodel/odata/odata-core/src/test/java/com/sap/cloud/sdk/datamodel/odata/helper/FunctionImportSingleDeserializationTest.java [hunk 6]
```diff
@@ -67,11 +67,13 @@ static Stream<TestInput<?>> testDeserializeSingleResponse()
             .of(
                 new TestInput<>(
                     String.class,
-                    "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
-                        + "<!DOCTYPE user-permission PUBLIC \"-//SuccessFactors, Inc.//DTD Permission Config//EN\" \"users-permissions.dtd\">\n"
-                        + "<user-permission>\n"
-                        + "<user id=\"sfadmin@SAPPHIRE17\"/>\n"
-                        + "</user-permission>\n",
+                    """
+                        <?xml version="1.0" encoding="UTF-8"?>
```
Comment (newtork) [inline datamodel/odata/odata-core/src/test/java/com/sap/cloud/sdk/datamodel/odata/helper/FunctionImportSingleDeserializationTest.java:71]: (Minor/Comment)

Tests seem to work, but I'm a little afraid of the 4 spaces prepending `<?xml` ... I would expect this to cause syntax error in regular XML parsers.

File: datamodel/odata/odata-core/src/test/java/com/sap/cloud/sdk/datamodel/odata/helper/ODataSerializationTest.java [hunk 7]
```diff
@@ -200,12 +192,9 @@ void testCreateResponse()
...
         final String SERIALIZED_ENTITY =
-            "{"
-                + "\"IntegerValue\":1,"
-                + "\"OffsetDateTimeValue\":\"/Date(694224000000-0240)/\","
-                + "\"LocalTimeValue\":\"PT13H20M0S\","
-                + "\"LocalDateTimeValue\":\"/Date(694224000000)/\""
-                + "}";
+            """
+                {"IntegerValue":1,"OffsetDateTimeValue":"/Date(694224000000-0240)/","LocalTimeValue":"PT13H20M0S","LocalDateTimeValue":"/Date(694224000000)/"}\
+                """;
```
Comment (newtork) [inline datamodel/odata/odata-core/src/test/java/com/sap/cloud/sdk/datamodel/odata/helper/ODataSerializationTest.java:197]: (Minor)

Escape new-lines, maybe :)

File: cloudplatform/cloudplatform-core/src/main/java/com/sap/cloud/sdk/cloudplatform/requestheader/RequestHeaderAccessor.java (hunk 1 above)
Comment (johannes) [reply cloudplatform/cloudplatform-core/src/main/java/com/sap/cloud/sdk/cloudplatform/requestheader/RequestHeaderAccessor.java:187]: > @johannes, what's your suggestion?

To be mindful in order to highlight the right person :-)

Have a good time! (No need to reply/apologize, I unsubscribed from this)

File: cloudplatform/cloudplatform-connectivity/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/EnvVarDestinationLoaderTest.java (hunk 2 above)
Comment (MatKuhr) [reply cloudplatform/cloudplatform-connectivity/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/EnvVarDestinationLoaderTest.java:99]: I applied some further manual improvements on the string texts of this test class and felt that these minor formatting changes on the assertions were close enough topic wise..

File: cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java (hunk 3 above)
Comment (MatKuhr) [reply cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java:46]: feel free to change it back in your PR if you feel it looks better 😉

File: datamodel/odata-client/src/test/java/com/sap/cloud/sdk/datamodel/odata/client/request/ODataResponseComplexDataParsingTest.java (hunk 5 above)
Comment (MatKuhr) [reply datamodel/odata-client/src/test/java/com/sap/cloud/sdk/datamodel/odata/client/request/ODataResponseComplexDataParsingTest.java:129]: Fixed and improved the formatting

File: datamodel/odata/odata-api-sample/src/test/java/com/sap/cloud/sdk/datamodel/odata/sample/ODataFetchMediaStreamTest.java [hunk 8]
```diff
@@ -39,30 +39,32 @@ class ODataFetchMediaStreamTest
...
-        "{"
-            + "\"d\" : {"
-            + "  \"results\" : ["
-            + "    {"
-            + "      \"__metadata\" : {"
-            + "        \"id\": \"https://127.0.0.1/com.sap.cloud.sdk.store.grocery/Products(44444)\"," // ignored by the SDK
-            + "        \"uri\": \"https://127.0.0.1/com.sap.cloud.sdk.store.grocery/Products(44444)\"," // ignored by the SDK
-            + "        \"type\": \"SdkGroceryStore.ProductType\","
-            + "        \"etag\": \"W/\\\"datetimeoffset'2023-04-18T10%3A55%3A00.5351290Z'\\\"\","
-            + "        \"content_type\" : \"text/plain\"," // ignored by the SDK
```
[the remark this answers is not in the document]
Comment (MatKuhr) [reply datamodel/odata/odata-api-sample/src/test/java/com/sap/cloud/sdk/datamodel/odata/sample/ODataFetchMediaStreamTest.java:51]: Yep, I now moved the comments around to keep them

**Review by newtork — APPROVED** [review]
